### ISSUE TITLE
improve scatter plot visibility

### ISSUE THREAD

**Issue by swharden** [swharden/ScottPlot#600]
When scatter plots have `visible = false` they should immediately return from `Render()` and they should also return empty limits. 

Together these improvements will allow scatter plots to be loaded but invisible, and not affect `AxisAuto()`.

**Comment by StendProg** [comment]
Hi @swharden,
I think `Visible` is the basic thing inherent in all Plottable. It is not correct to implement it separately in each class from the point of view of code duplication.
There are two possible solutions:
1. Filter the Plottable list when calling them `Render` and `GetLimits` taking into account the `Visibility` flag.
2. Move the common functionality to the base class PlottableBase.

From a performance point of view, option 1 is preferable. 2 will have a more structured code.

**Comment by swharden** [comment]
Thanks for your input @StendProg! This gave me a lot to think about while I refactored over the last few weeks. After refactoring a lot of modules and running into this problem many times, I found that I prefer putting visibility logic inside the 
Render method (returning immediately if the visible flag is false).

This strategy may be slightly less performant, but I think it is a good trade: We lose a few CPU cycles, but gain an **easy to read calling function** which does not contain `Select().Where()` or `if` statements. This is especially true when rendering axes, which have major and minor ticks, major and minor grid lines, axis labels, tick labels, and a frame edge (each with its own visibility flag). That's a lot of `if` statements to keep track of.

I also found it helpful to not worry too much about micro-optimizations because the line drawing and area filling methods to render the final image take a lot of time compared to small things like a few extra method calls.

(Also there are some new interfaces which address the axis limits issue described earlier)
